internal/tls/tls_test.go

```diff
@@ -53,36 +56,132 @@ func TestNew(t *testing.T) {
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           opts: nil,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           opts: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "returns cfg and nil when option is not empty",
+			args: args{
+				opts: []Option{
+					WithCert("./testdata/dummyServer.crt"),
+					WithKey("./testdata/dummyServer.key"),
+					WithCa("./testdata/dummyCa.pem"),
+				},
+			},
+			want: want{
+				want: func() *Config {
+					var err error
+					c := &credentials{
+						cfg: new(tls.Config),
+					}
+
+					c.cfg.Certificates = make([]tls.Certificate, 1)
+					c.cfg.Certificates[0], err = tls.LoadX509KeyPair("./testdata/dummyServer.crt", "./testdata/dummyServer.key")
+					if err != nil {
+						panic(err)
+					}
+
+					pool := x509.NewCertPool()
+					b, err := ioutil.ReadFile("./testdata/dummyCa.pem")
+					if err != nil {
+						panic(err)
```

Comment:
🤔 
I think testing.T is better...:thinking:

Reply:
for example,

```
type want struct {
    wantFunc: func(t *tesitng.T) want {

    }
}

```

Reply:
should we ignore the error? 
```
b, _ := ioutil.ReadFile("./testdata/dummyCa.pem")
```

Reply:
it seems good, If absolutely guaranteed to exist!

Reply:
thank you!